backend/app/services/price_fetcher.py

```
@@ -321,20 +221,22 @@ def fetch_latest_price(self, ticker: str, isin: Optional[str] = None) -> Optiona
-                return asyncio.run(self.fetch_crypto_price(ticker))
-            else:
-                # Resolve ticker using ISIN if provided
-                resolved_ticker = TickerMapper.resolve_ticker(ticker, isin)
-                logger.info(f"Fetching price for {ticker} (resolved: {resolved_ticker})")
+            # Resolve ticker using ISIN if provided
+            resolved_ticker = TickerMapper.resolve_ticker(ticker, isin) if isin else ticker
+            logger.info(f"Fetching price for {ticker} (resolved: {resolved_ticker})")
+            
+            # Check if we're in an async context
+            try:
+                loop = asyncio.get_running_loop()
```

Ruff (0.14.0)

Local variable `loop` is assigned to but never used

Remove assignment to unused variable `loop`

(F841)